File: web/src/test/java/top/bettercode/summer/test/web/support/setting/CorsProperties.java
package top.bettercode.summer.test.web.support.setting;

import java.util.Collections;
import org.springframework.web.cors.CorsConfiguration;

/**
 * @author Peter Wu
 * @since 0.1.12
 */
public class CorsProperties extends CorsConfiguration {

  private boolean enable = true;
  private String path = "/**";

  public CorsProperties() {
    setAllowedOrigins(Collections.singletonList("*"));
    setAllowedMethods(Collections.singletonList("*"));
    //    setAllowCredentials(true);
    setAllowedHeaders(Collections.singletonList("*"));
  }

  public boolean isEnable() {
    return enable;
  }

  public void setEnable(boolean enable) {
    this.enable = enable;
  }

  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/properties/CorsProperties.kt
package top.bettercode.summer.web.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.web.cors.CorsConfiguration

/**
 * @author Peter Wu
 * @since 0.1.12
 */
@ConfigurationProperties("summer.security.cors")
open class CorsProperties : CorsConfiguration() {
    var isEnable = true
    var path = "/**"

    init {
        allowedOrigins = listOf(ALL)
        allowedMethods = listOf(ALL)
        //    setAllowCredentials(true);
        allowedHeaders = listOf(ALL)
    }
}
